src/v/kafka/client/consumer.h

```diff
@@ -107,9 +111,18 @@ class consumer final : public ss::enable_lw_shared_from_this<consumer> {
         using api_t = typename std::invoke_result_t<RequestFactory>::api_type;
         using response_t = typename api_t::response_type;
         return ss::try_with_gate(_gate, [this, req{std::move(req)}]() {
-            auto r = req();
-            kclog.debug("Consumer: {}: {} req: {}", *this, api_t::name, r);
-            return _coordinator->dispatch(std::move(r))
+            auto f = ss::now();
+            if (!_coordinator) {
+                f = find_coordinator().then(
```

Comment:
@BenPope I'm wondering if this could be an issue since, in the case mitigation fails, this will throw

Reply:
I think I'd ensure that `_coordinator` exists during `join()` - so reqork the new stuff in `initialize` into join().

Reply:
Actually I'm not sure that would make any real difference, would it? Are other req_res happening before join?

Reply:
Yes that is why I did that. Public methods like `leave()` for example will call `req_res` and if `_coordinator` hasn't been initialized then it would be nullptr.

But I guess this is only an issue if someone constructs a `consumer` instance and doesn't call `initialize()` which, they could since the constructor is public, but obviously wouldn't because there is a `make_consumer()` method which handles initialization for you.

Thinking about it again, I think I should remove this, thoughts?

Reply:
I don't think this is needed. `make_consumer()`  should be sufficient to guard against a null `_consumer`.